Can no longer run clippy just for layout_2020

I mostly work on layout_2020, so I would run `./mach clippy -r -p layout_2020` to only check clippy there.

But now I get

```
$ ./mach clippy -r -p layout_2020
    Checking net_traits v0.0.1 (/home/oriol/src/servo/components/shared/net)
error[E0432]: unresolved import `hyper_util::client::legacy::Error`
  --> components/shared/net/lib.rs:19:5
   |
19 | use hyper_util::client::legacy::Error as HyperError;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `Error` in `client::legacy`
   |
note: found an item that was configured out
  --> /home/oriol/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hyper-util-0.1.10/src/client/legacy/mod.rs:4:35
   |
4  | pub use client::{Builder, Client, Error, ResponseFuture};
   |                                   ^^^^^
note: the item is gated here
  --> /home/oriol/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hyper-util-0.1.10/src/client/legacy/mod.rs:3:1
   |
3  | #[cfg(any(feature = "http1", feature = "http2"))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: consider importing one of these items instead
   |
19 | use crate::ResourceTimingType::Error as HyperError;
   |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
19 | use crate::request::BodyChunkRequest::Error as HyperError;
   |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
19 | use crate::request::BodyChunkResponse::Error as HyperError;
   |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
19 | use crate::request::RedirectMode::Error as HyperError;
   |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     and 17 other candidates

For more information about this error, try `rustc --explain E0432`.
error: could not compile `net_traits` (lib) due to 1 previous error
```

It works well with `./mach clippy -r -p net -p layout_2020`.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -72,7 +72,7 @@ http-body-util = "0.1"
 hyper = "1.0"
 hyper-rustls = { version = "0.27", default-features = false, features = ["http1", "http2", "logging", "tls12", "webpki-tokio"] }
 hyper_serde = { path = "components/hyper_serde" }
-hyper-util = "0.1"
+hyper-util = { version = "0.1", features = ["client", "client-legacy", "http2", "tokio"] }
 icu_segmenter = "1.5.0"
 image = "0.24"
 imsz = "0.2"

--- a/components/net/Cargo.toml
+++ b/components/net/Cargo.toml
@@ -37,7 +37,7 @@ http-body-util = { workspace = true }
 hyper = { workspace = true, features = ["client", "http1", "http2"] }
 hyper-rustls = { workspace = true }
 hyper_serde = { workspace = true }
-hyper-util = { workspace = true, features = ["client", "client-legacy", "tokio"] }
+hyper-util = { workspace = true }
 imsz = { workspace = true }
 ipc-channel = { workspace = true }
 log = { workspace = true }

--- a/components/shared/net/Cargo.toml
+++ b/components/shared/net/Cargo.toml
@@ -23,7 +23,7 @@ headers = { workspace = true }
 http = { workspace = true }
 hyper = { workspace = true }
 hyper_serde = { workspace = true }
-hyper-util = { workspace = true, features = ["client-legacy"] }
+hyper-util = { workspace = true }
 ipc-channel = { workspace = true }
 log = { workspace = true }
 malloc_size_of = { workspace = true }

